### PR TITLE
[MOSIP-44079]: Junit test cases for </oidc-clients/{clientId}> GET,PUT and PATCH API's

### PR DESCRIPTION
**Pull request by chetankh239**
## Summary by CodeRabbit

* **Tests**
  * Expanded and hardened test coverage for OIDC v2 client management (GET, UPDATE, DEACTIVATE) covering success paths, feature-disabled scenarios, validation failures, invalid/empty/null IDs, client-not-found, additional-config and policy-edge cases.
  * Standardized many test method names to camelCase and adjusted test scaffolding to improve dependency injection and maintainability, plus numerous end-to-end style test additions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java [hunk 1]
```diff
@@ -532,4 +532,49 @@ public void getPartnersClientsV2TestwithoutFilters() throws Exception {
...
+    public void testGetOIDCClientV2featureDisabled() throws Exception {
+        String clientId = "test-client-123";
+
+        doThrow(new PartnerServiceException("PMS_OIDC_001", "OIDC client feature is not enabled"))
+                .when(featureAvailabilityUtil).validateOidcClientFeatureEnabled();
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/oidc-clients/{clientId}", clientId)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+    }
```
[the remark this answers is not in the document]
Comment (chetankh239) [reply partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java:837]: done changes

File: partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java [hunk 2]
```diff
@@ -532,4 +532,49 @@ public void getPartnersClientsV2TestwithoutFilters() throws Exception {
...
+    @WithMockUser(roles = {"AUTH_PARTNER"})
+    public void testGetOIDCClientV2invalidClientId() throws Exception {
+        String invalidClientId = "invalid~client~id";
+
+        doNothing().when(featureAvailabilityUtil).validateOidcClientFeatureEnabled();
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/oidc-clients/{clientId}", invalidClientId)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+    }
```
[the remark this answers is not in the document]
Comment (chetankh239) [reply partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java:565]: done changes

File: partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chetankh239) [reply partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java:837]: > @chetankh239, understood. I've noted your decision to skip this change.
> 
> 🧠 Learnings used

not to skip, verify the changes

[the remark this answers is not in the document]
Comment (chetankh239) [reply partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java:837]: done changes

**Review by mayuradesh — APPROVED** [review]

File: partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java [hunk 3]
```diff
@@ -532,4 +532,307 @@ public void getPartnersClientsV2TestwithoutFilters() throws Exception {
...
+        requestWrapper.setVersion("1.0");
+
+        doNothing().when(featureAvailabilityUtil).validateOidcClientFeatureEnabled();
+        doNothing().when(featureAvailabilityUtil).validateOidcClientAdditionalInfoFeatureAvailable();
+
+        String requestJson = objectMapper.writeValueAsString(requestWrapper);
+        mockMvc.perform(MockMvcRequestBuilders.patch("/oidc-clients/{clientId}", invalidClientId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestJson));
+    }
```
[the remark this answers is not in the document]
Comment (chetankh239) [reply partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java:767]: can be ignored

File: partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java [hunk 4]
```diff
@@ -532,4 +532,307 @@ public void getPartnersClientsV2TestwithoutFilters() throws Exception {
...
+        requestWrapper.setVersion("1.0");
+
+        doThrow(new PartnerServiceException("PMS_OIDC_001", "OIDC client feature is not enabled"))
+                .when(featureAvailabilityUtil).validateOidcClientFeatureEnabled();
+
+        String requestJson = objectMapper.writeValueAsString(requestWrapper);
+        mockMvc.perform(MockMvcRequestBuilders.patch("/oidc-clients/{clientId}", clientId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestJson));
+    }
```
[the remark this answers is not in the document]
Comment (chetankh239) [reply partner/partner-management-service/src/test/java/io/mosip/pms/test/oauth/controller/ClientManagementControllerTest.java:837]: Not affect, can be ignored

**Review by mayuradesh — APPROVED** [review]
